components/script/dom/bindings/str.rs

```diff
@@ -186,6 +186,30 @@ impl DOMString {
     pub fn bytes(&self) -> Bytes {
         self.0.bytes()
     }
+
+    /// Removes newline characters according to <https://infra.spec.whatwg.org/#strip-newlines>.
+    pub fn strip_newlines(&mut self) {
+        self.0.retain(|c| c != '\r' && c != '\n');
+    }
+
+    /// Removes leading and trailing ASCII whitespaces according to
+    /// <https://infra.spec.whatwg.org/#strip-leading-and-trailing-ascii-whitespace>.
+    pub fn strip_leading_and_trailing_ascii_whitespace(&mut self) {
+        if self.0.len() == 0 { return; }
+
+        let last_non_whitespace = match self.0.rfind(|c| !char::is_whitespace(c)) {
```

Comment:
This is still incorrect, this should be about ASCII whitespace, not all whitespace.